packages/contracts/contracts/test/AssetManager.t.sol

```diff
@@ -136,514 +79,781 @@ contract AssetManagerTest is ValidatorSystemUpgrade_Initializer {
-        assertEq(assetManager.UNDELEGATION_PERIOD(), undelegationPeriod);
-        assertEq(assetManager.SLASHING_RATE(), slashingRate);
-        assertEq(assetManager.MIN_SLASHING_AMOUNT(), minSlashingAmount);
-    }
-
-    function test_constructor_largeSlashingRate_reverts() external {
-        vm.expectRevert(IAssetManager.InvalidConstructorParams.selector);
-        new MockAssetManager(
-            kro,
-            kgh,
-            kghManager,
-            address(guardian),
-            valMgr,
-            uint128(undelegationPeriod),
-            1001,
-            minSlashingAmount
-        );
+        assertEq(address(assetMgr.ASSET_TOKEN()), address(assetToken));
+        assertEq(address(assetMgr.KGH()), address(kgh));
+        assertEq(assetMgr.SECURITY_COUNCIL(), guardian);
+        assertEq(address(assetMgr.VALIDATOR_REWARD_VAULT()), validatorRewardVault);
+        assertEq(address(assetMgr.VALIDATOR_MANAGER()), address(valMgr));
+        assertEq(assetMgr.MIN_DELEGATION_PERIOD(), minDelegationPeriod);
+        assertEq(assetMgr.BOND_AMOUNT(), bondAmount);
     }
 
-    function test_delegate_succeeds() external {
-        _setUpKroDelegation(100e18);
+    function test_depositToRegister_succeeds() external {
+        uint256 beforeBalance = assetToken.balanceOf(validator);
+
+        vm.prank(validator);
+        assetToken.approve(address(assetMgr), minActivateAmount);
+        vm.prank(address(valMgr));
+        assetMgr.depositToRegister(validator, minActivateAmount, withdrawAcc);
 
-        assertEq(assetManager.totalKroAssets(validator), 100e18);
-        assertEq(valMgr.getWeight(validator), 200e18);
-        assertEq(assetManager.totalValidatorKro(validator), 100e18);
+        assertEq(assetMgr.getWithdrawAccount(validator), withdrawAcc);
+        assertEq(assetMgr.totalValidatorKro(validator), minActivateAmount);
+        assertEq(assetMgr.canWithdrawAt(validator), block.timestamp + minDelegationPeriod);
+        assertEq(assetToken.balanceOf(validator), beforeBalance - minActivateAmount);
+        assertEq(assetToken.balanceOf(address(assetMgr)), minActivateAmount);
     }
 
-    function test_delegate_withoutValidatorDelegation_reverts() external {
-        vm.expectRevert(IAssetManager.ImproperValidatorStatus.selector);
-        assetManager.delegate(validator, 100e18);
+    function test_depositToRegister_callerNotValMgr_reverts() external {
+        vm.expectRevert(IAssetManager.NotAllowedCaller.selector);
+        assetMgr.depositToRegister(validator, minActivateAmount, withdrawAcc);
     }
 
-    function test_delegateKgh_succeeds() external {
-        (uint128 kroShares, uint128 kghShares) = _setUpKghDelegation(1);
+    function test_depositToRegister_zeroAsset_reverts() external {
+        vm.prank(address(valMgr));
+        vm.expectRevert(IAssetManager.NotAllowedZeroInput.selector);
+        assetMgr.depositToRegister(validator, 0, withdrawAcc);
+    }
```

Comment:
How do you think about removing this test? Since if PR #360 is merged, this test will fail.

Reply:
Sure, I'll remove this.